instrumentation/spring/spring-security-config-6.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/security/config/v6_0/servlet/SpringSecurityConfigServletInstrumentationModule.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.security.config.v6_0.servlet;
+
+import static io.opentelemetry.javaagent.extension.matcher.AgentElementMatchers.hasClassesNamed;
+import static java.util.Collections.singletonList;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.bootstrap.internal.CommonConfig;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.util.List;
+import net.bytebuddy.matcher.ElementMatcher;
+
+/** Instrumentation module for servlet-based applications that use spring-security-config. */
+@AutoService(InstrumentationModule.class)
+public class SpringSecurityConfigServletInstrumentationModule extends InstrumentationModule {
+  public SpringSecurityConfigServletInstrumentationModule() {
+    super("spring-security-config-servlet", "spring-security-config-servlet-6.0");
+  }
+
+  @Override
+  public boolean defaultEnabled(ConfigProperties config) {
+    /*
+     * Since the only thing this module currently does is capture enduser attributes,
+     * the module can be completely disabled if enduser attributes are disabled.
+     *
+     * If any functionality not related to enduser attributes is added to this module,
+     * then this check will need to move elsewhere to only guard the enduser attributes logic.
+     */
+    return CommonConfig.get().getEnduserConfig().isAnyEnabled();
+  }
+
+  @Override
+  public ElementMatcher.Junction<ClassLoader> classLoaderMatcher() {
+    return hasClassesNamed(
```

Comment:
Is this really needed? We usually use `classLoaderMatcher` when there are multiple instrumentation fro the same library that should be used with different versions of the library.

Reply:
Good question.  This might be handled properly muzzle's runtime checks.

I wanted to explicitly ensure that these three libraries were on the classpath for this module to be enabled...
* spring-security-config (HttpSecurity) - contains the class being instrumented
* spring-security-web (AuthorizationFilter) - used by EnduserAttributesHttpSecurityCustomizer
* jakarta.servlet-api (jakarta.servlet.Servlet) - used by EnduserAttributesCapturingServletFilter

If muzzle's runtime checks will guarantee that, then I can remove this method.

The main thing I want to ensure is that it does not apply to spring-security < 6.0, since that uses javax.servlet.   I'm not even sure if that is properly handled with the given matcher.

Reply:
I changed it to check for a class that was added in spring-security 6.0